posts/frangeris/02.reverse-proxy-cache-using-cloudfront.md

```diff
@@ -0,0 +1,145 @@
+---
+title: Reverse proxy cache using AWS CloudFront
+published: false
+description: Detailed explanation on how to implement a reverse proxy cache to any API using AWS CloudFront
+tags: "tutorial, aws, cache, cloudfront"
+cover_image: ./assets/02.cover.png
+---
+
+## How to implement a reverse proxy cache to any API using AWS CloudFront
+
+> A reverse proxy is the application that sits in front of back-end applications and forwards client (e.g. browser) requests to those applications. Reverse proxies help increase scalability, performance, resilience and security. The resources returned to the client appear as if they originated from the web server itself.
+
+AWS CloudFront is a CDN service for high performance and security convenience that offers a lot of advantages including a global edge network with a low latency and high throughput network connectivity (the one that matter to us).
+
+One typical example where we could be needing a reverse proxy cache mechanism is when building HTTP APIs (Api Gateway V2) on AWS, this type of APIs are designed with minimal features so that they can be offered at lower price, lacking options as edge optimization, support for api keys, throttling and cache, [more detailed comparison here](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-vs-rest.html); not having support for cache means processing time load will increase on backend side on origin servers, resulting in high latency on every request.
```

Comment:
```suggestion
One typical example where we could be needing a reverse proxy cache mechanism is when building HTTP APIs (API Gateway v2) on AWS, this type of APIs are designed with minimal features so that they can be offered at lower price, lacking options as edge optimization, support for api keys, throttling and cache, [more detailed comparison here](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-vs-rest.html); not having support for cache means processing time load will increase on backend side on origin servers, resulting in high latency on every request.
```